hpx/plugins/plugin_registry.hpp

```diff
@@ -89,15 +87,32 @@ namespace hpx { namespace plugins
     )(__VA_ARGS__))                                                           \
 /**/
 
-#define HPX_REGISTER_PLUGIN_REGISTRY_2(PluginType, pluginname)                \
-    typedef hpx::plugins::plugin_registry<PluginType>                         \
-        pluginname ## _plugin_registry_type;                                  \
-    HPX_REGISTER_PLUGIN_BASE_REGISTRY(                                        \
-        pluginname ## _plugin_registry_type, pluginname)                      \
-    HPX_DEF_UNIQUE_PLUGIN_NAME(                                               \
-        pluginname ## _plugin_registry_type, pluginname)                      \
-    template struct hpx::plugins::plugin_registry<PluginType >;               \
-/**/
+#define HPX_REGISTER_PLUGIN_REGISTRY_2(PluginType, pluginname)                 \
+    HPX_REGISTER_PLUGIN_REGISTRY_5(                                            \
+        PluginType, pluginname, HPX_PLUGIN_NAME, "hpx", "hpx")                 \
+    /**/
+#define HPX_REGISTER_PLUGIN_REGISTRY_4(                                        \
+    PluginType, pluginname, pluginsection, pluginsuffix)                       \
+    HPX_REGISTER_PLUGIN_REGISTRY_5(PluginType, pluginname, HPX_PLUGIN_NAME,    \
+        pluginsection, "pluginsuffix")                                         \
+    /**/
+#define HPX_REGISTER_PLUGIN_REGISTRY_5(                                        \
+        PluginType, pluginname, pluginstring, pluginsection, pluginsuffix)     \
+    HPX_CXX14_CONSTEXPR char __##pluginname##_string[] =                       \
+        HPX_PP_STRINGIZE(pluginstring);                                        \
+    HPX_CXX14_CONSTEXPR char __##pluginname##_section[] = pluginsection;       \
+    HPX_CXX14_CONSTEXPR char __##pluginname##_suffix[] = pluginsuffix;         \
+    typedef hpx::plugins::plugin_registry<PluginType, __##pluginname##_string, \
+        __##pluginname##_section, __##pluginname##_suffix>                     \
+        __##pluginname##_plugin_registry_type;                                 \
+    HPX_REGISTER_PLUGIN_BASE_REGISTRY(                                         \
+        __##pluginname##_plugin_registry_type, pluginname)                     \
+    HPX_DEF_UNIQUE_PLUGIN_NAME(                                                \
+        __##pluginname##_plugin_registry_type, pluginname)                     \
+    template struct hpx::plugins::plugin_registry<PluginType,                  \
+        __##pluginname##_string, __##pluginname##_section,                     \
+        __##pluginname##_suffix>;                                              \
```

Comment:
Should we rather use `HPX_PP_CAT` to concatenate the PP tokens?

Reply:
This is only necessary if the parameters `pluginname` is a macro itself.. Not sure if it's worth the trouble.

Reply:
Ok, we'll see how it goes.